frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -88,6 +88,12 @@ function ReaderTypeset:onReadSettings(config)
     self.ui.document:setNightmodeImages(self.configurable.nightmode_images == 1)
 end
 
+function ReaderTypeset:onReaderReady()
+    -- Initial detecting of fb2 may be wrong
+    local doc_format = self.ui.document:getDocumentFormat()
+    self.ui.document.is_fb2 = doc_format:find("FB") and true
```

Comment:
May be keep the string search consistent with what you use below ?
`file_type:sub(1, 2) == "fb"`
:sub instead of :find, so you survive the day when we start supporting KFB (Kindle Fancy Book :))

Reply:
getDocumentFormat() returns for fb2 "FictionBook (FB2)"

Reply:
I have no fb3 to test, suppose the similar.

Reply:
Oh, ok, so it doesn't use the stuff I mentionned (which seems not used at all...), but it uses:
https://github.com/koreader/crengine/blob/1aa6b9ecfd4d1ad0d7013ad72fab60f16691f7a7/crengine/src/lvdocview.cpp#L4840-L4856
So, OK.
(Or may be `doc_format:sub(1, 11) == "FictionBook"`.)

Reply:
On the first opening, if initial fb2 detection was wrong, maybe we should reapply setStyleSheet with the correct css?

Reply:
Maybe, as you feel, dunno how complicated it is and if it would cause a rerender.
I remember I do set some styletweak in my own patch with an added `ReaderStyleTweak:onPreRenderDocument()`, so checking/fixing (if we know the document type at this point) in a onPreRenderDocument() may avoid a rerendering.

Reply:
https://github.com/koreader/koreader/issues/12741 is not fixed yet. @mergen3107 gets `fb2.css` Style menu item enabled, and must choose it manually.

Reply:
OK, so if you think this is the solution, go for it.

Reply:
Done